### PR TITLE
Functional tests: allow `grep_fail` to accept options

### PR DESCRIPTION
**Pull request by MetRonnie**
Like the description says. Split off from #6370 

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] No dependency changes 
- [x] No changelog/docs needed as internal
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: tests/functional/cli/05-colour.t [hunk 1]
```diff
@@ -23,16 +23,19 @@
...
 script -q -c "cylc scan -t rich" log > /dev/null 2>&1
 grep_ok "$ANSI" log -P  # color
 
-script -q -c "cylc scan -t rich --color=never" log > /dev/null 2>&1
-grep_fail "$ANSI" log -P  # no color
+# FIXME: this test doesn't work because of some interaction between the
+# `script` command and colorama. The output includes a color reset char at
+# the end despite using colorama.init(autoreset=False).
+# script -q -c "cylc scan -t rich --color=never" log > /dev/null 2>&1
+# grep_fail "$ANSI" log -P  # no color
```
Comment (oliver-sanders) [inline tests/functional/cli/05-colour.t:38]: This test wasn't testing what it was supposed to test before?

Comment (oliver-sanders) [reply tests/functional/cli/05-colour.t:38]: Ok, we should link an issue to this FIXME.

Would this close it: https://github.com/cylc/cylc-flow/issues/6076 ?

Comment (MetRonnie) [reply tests/functional/cli/05-colour.t:38]: No (I tried turning autoreset back on even though that wouldn't make any sense for it to fix the test, and it didn't help).

---

N.B. I had a look at how colorama works. It bills itself as a package meant only for making colour work on Windows, but I guess we are using it for its `init(strip=not use_color)` function that can automatically strip ANSI color chars from `print()` calls (it actually patches `sys.stdout` and `sys.stderr`). However there are >100 open issues and it is in a call-for-maintainers state: https://github.com/tartley/colorama/issues/300.

Comment (MetRonnie) [reply tests/functional/cli/05-colour.t]: Link to #6467 added

**Review by hjoliver — APPROVED** [review]
Looks fine in terms of grep_fail options, question about the disabled test aside.

**Comment by MetRonnie** [comment]
Merging with 1.5 approvals...

**Pull request merged**